openapi.yaml

```diff
@@ -1,279 +1,684 @@
-swagger: '2.0'
+openapi: 3.0.0
 info:
-  title: Express API Skeleton
-  description: Skeleton for Express APIs
-  version: v1
+  title: Locations API
+  description: Get information about OSU locations and services
+  version: v2
   license:
     name: GNU Affero General Public License Version 3
     url: http://www.gnu.org/licenses/agpl-3.0.en.html
   contact:
     name: IS Data Architecture Team
     url: https://is.oregonstate.edu/data-architecture
     email: isdataarchitecture@oregonstate.edu
-schemes:
-  - https
-host: api.oregonstate.edu
-basePath: /v1
 externalDocs:
   description: GitHub Repository
-  url: https://github.com/osu-mist/express-api-skeleton
-produces:
-  - application/json
-consumes:
-  - application/json
+  url: https://github.com/osu-mist/locations-frontend-api
+servers:
+  - url: https://api.oregonstate.edu/v2
 security:
   - OAuth2:
       - full
 paths:
-  /pets:
+  /locations:
     get:
-      summary: List all pets
       tags:
-        - pets
-      description: Get all pets
-      operationId: getPets
+       - locations
+      summary: Get locations
+      description: Gets all `Location` objects. Can be filtered by parameters.
+      operationId: getLocations
       parameters:
-        - $ref: '#/parameters/pageNumber'
-        - $ref: '#/parameters/pageSize'
         - in: query
-          name: species
-          description: Search by pet species
+          name: filter[search]
+          description: 'Search through buildings based on building name and abbreviation'
           required: false
-          type: string
+          schema:
+            type: string
+        - in: query
+          name: filter[latitude]
+          description: 'Latitude of the location. Should be used with `filter[longitude]` parameter'
+          required: false
+          schema:
+            type: number
+            format: float
+        - in: query
+          name: filter[longitude]
+          description: 'Longitude of the location. Should be used with `filter[latitude]` parameter'
+          required: false
+          schema:
+            type: number
+            format: float
```

Comment:
As Liang mentioned, longitude and latitude must be used together. The only way that comes to mind where we could enforce this in this document is if we grouped these two parameters into one `filter[coordinates]` param which takes an array of the latitude and longitude.

Reply:
oops, just saw that you already addressed this